execution/registry/registry.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2017 Monax Industries Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hyperledger/burrow/crypto"
+	"github.com/tendermint/tendermint/p2p"
+)
+
+// NewRegisteredNode takes a p2p node key, and returns a node entry with a signature
+func NewRegisteredNode(nodeKey *p2p.NodeKey, moniker, netAddress string) (*RegisteredNode, []byte, error) {
+	burrowNodePubKey, err := crypto.PublicKeyFromTendermintPubKey(nodeKey.PubKey())
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not convert tm publicKey: %v", err)
+	}
+
+	burrowNodeID, err := crypto.AddressFromHexString(string(nodeKey.ID()))
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not convert tm nodeID: %v", err)
+	}
+
+	node := &RegisteredNode{
+		Moniker:    moniker,
+		NetAddress: netAddress,
+		ID:         burrowNodeID,
+		PublicKey:  burrowNodePubKey,
+	}
+
+	data, err := node.Marshal()
```

Comment:
Again this should be deterministic protobuf.

Alternatively, we could sign the JSON serialisation of this object. I have done this elsewhere with the justification that it is probably easier for clients to generate JSON than protobuf.

In this case I suppose we normally control both ends, but I could imagine wanting to send the signed payload from a JS client perhaps before I start a node. What do you think?